checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1708 @@
+   *     formal parameter (or the receiver), returns the result of calling {@link
+   *     #removeCastsAndGetTmpVarIfPresent(Node)} on the actual parameter passed in that position.
+   *     Otherwise, returns {@code null}.
+   */
+  private @Nullable Node getMustCallAliasParamVar(Node callNode) {
+    Preconditions.checkArgument(
+        callNode instanceof MethodInvocationNode || callNode instanceof ObjectCreationNode);
+    if (!typeFactory.hasMustCallAlias(callNode.getTree())) {
+      return null;
+    }
+
+    Node result = null;
+    List<Node> actualParams = getArgumentsOfInvocation(callNode);
+    List<? extends VariableElement> formalParams = getFormalsOfInvocation(callNode);
+    for (int i = 0; i < actualParams.size(); i++) {
+      if (typeFactory.hasMustCallAlias(formalParams.get(i))) {
+        result = actualParams.get(i);
+        break;
+      }
+    }
+
+    // If none of the parameters were @MustCallAlias, it must be the receiver
+    if (result == null && callNode instanceof MethodInvocationNode) {
+      result = ((MethodInvocationNode) callNode).getTarget().getReceiver();
+    }
+
+    result = removeCastsAndGetTmpVarIfPresent(result);
+    return result;
+  }
+
+  /**
+   * If a temporary variable exists for node after typecasts have been removed, return it.
+   * Otherwise, return node.
+   *
+   * @param node a node
+   * @return either a tempvar for node's content sans typecasts, or node
+   */
+  private Node removeCastsAndGetTmpVarIfPresent(Node node) {
+    // TODO create temp vars for TypeCastNodes as well, so we don't need to explicitly remove casts
+    // here
+    node = removeCasts(node);
+    LocalVariableNode tmpVar = typeFactory.getTempVarForNode(node);
+    return tmpVar != null ? tmpVar : node;
+  }
+
+  /**
+   * Get the nodes representing the arguments of a method or constructor invocation from the
+   * invocation node.
+   *
+   * @param node a MethodInvocation or ObjectCreation node
+   * @return a list of the arguments, in order
+   */
+  private List<Node> getArgumentsOfInvocation(Node node) {
+    List<Node> arguments;
+    if (node instanceof MethodInvocationNode) {
+      MethodInvocationNode invocationNode = (MethodInvocationNode) node;
+      arguments = invocationNode.getArguments();
+    } else {
+      if (!(node instanceof ObjectCreationNode)) {
+        throw new BugInCF("unexpected node type " + node.getClass());
+      }
+      arguments = ((ObjectCreationNode) node).getArguments();
+    }
+    return arguments;
+  }
+
+  /**
+   * Get the elements representing the formal parameters of a method or constructor, from an
+   * invocation of that method or constructor.
+   *
+   * @param node a method invocation or object creation node
+   * @return a list of the declarations of the formal parameters of the method or constructor being
+   *     invoked
+   */
+  private List<? extends VariableElement> getFormalsOfInvocation(Node node) {
```

Comment:
Both "formals" and "parameters" are acceptable terminology for formal parameters.  However, given that ExecutableElement (which this calls) has `getParameters`, I think naming this `getParametersOfInvocation` makes more sense.  It's good to try to use the same terminology for the same concept.